src/spi.rs

```diff
@@ -55,6 +56,85 @@ where
 {
 }
 
+/// A structure for specifying SPI configuration.
+///
+/// This structure uses builder semantics to generate the configuration.
+///
+/// `Example`
+/// ```
+/// use embedded_hal::spi::Mode;
+///
+/// let config = Config::new(Mode::MODE_0)
+///     .frame_size(8)
+///     .freeze();
+/// ```
+#[derive(Copy, Clone)]
+pub struct Config {
+    mode: Mode,
+    swap_miso_mosi: bool,
+    cs_delay: f32,
+    frame_size: u8,
+}
+
+impl Config {
+    /// Create a default configuration for the SPI interface.
+    ///
+    /// Arguments:
+    /// * `mode` - The SPI mode to configure.
+    pub fn new(mode: Mode) -> Self {
+        Config {
+            mode: mode,
+            swap_miso_mosi: false,
+            cs_delay: 0.0,
+            frame_size: 8_u8,
+        }
+    }
+
+    /// Specify that the SPI MISO/MOSI lines are swapped.
+    ///
+    /// Note:
+    /// * This function updates the HAL peripheral to treat the pin provided in the MISO parameter
+    /// as the MOSI pin and the pin provided in the MOSI parameter as the MISO pin.
+    pub fn swap_mosi_miso(&mut self) -> &mut Self {
```

Comment:
I think you can just say `pub fn swap_mosi_miso(self) -> Self {...}`. That removes the need for a `freeze()` method

Reply:
Should be fixed now to not use `freeze()`